connectors/src/connectors/confluence/temporal/activities.ts

```diff
@@ -0,0 +1,284 @@
+  const connector = await Connector.findOne({
+    where: {
+      type: "confluence",
+      id: connectorId,
+    },
+  });
+  if (!connector) {
+    throw new Error("Could not find connector");
+  }
+
+  const res = await syncSucceeded(connector.id);
+  if (res.isErr()) {
+    throw res.error;
+  }
+}
+
+export async function confluenceGetSpaceNameActivity({
+  confluenceCloudId,
+  connectionId,
+  spaceId,
+}: {
+  confluenceCloudId: string;
+  connectionId: string;
+  spaceId: string;
+}) {
+  const client = await getConfluenceClient({
+    cloudId: confluenceCloudId,
+    connectionId: connectionId,
+  });
+
+  const space = await client.getSpaceById(spaceId);
+
+  return space.name;
+}
+
+export async function confluenceListPageIdsInSpaceActivity({
+  confluenceCloudId,
+  connectionId,
+  pageCursor,
+  spaceId,
+}: {
+  confluenceCloudId: string;
+  connectionId: string;
+  pageCursor: string;
+  spaceId: string;
+}) {
+  const localLogger = logger.child({
+    spaceId,
+    pageCursor,
+  });
+
+  const client = await getConfluenceClient({
+    cloudId: confluenceCloudId,
+    connectionId: connectionId,
+  });
+
+  localLogger.info("Fetching Confluence pages in space.");
+
+  const { pages, nextPageCursor } = await client.getPagesInSpace(
+    spaceId,
+    pageCursor
+  );
+
+  return {
+    pageIds: pages.map((p) => p.id),
+    nextPageCursor,
+  };
+}
+
+async function upsertConfluencePageInDb(
+  connectionId: string,
+  dataSourceConfig: DataSourceConfig,
+  page: ConfluencePageWithBodyType
+) {
+  const connector = await Connector.findOne({
+    where: {
+      connectionId,
+      dataSourceName: dataSourceConfig.dataSourceName,
+      workspaceId: dataSourceConfig.workspaceId,
+    },
+  });
+
+  if (!connector) {
+    throw new Error(`Connector not found (connectionId: ${connectionId})`);
+  }
+
+  await ConfluencePage.upsert({
+    connectorId: connector.id,
+    pageId: page.id,
+    spaceId: page.spaceId,
+    parentId: page.parentId,
+    title: page.title,
+    externalUrl: page._links.tinyui,
+    version: page.version.number,
+  });
+}
+
+interface ConfluenceUpsertPageActivityInput {
+  spaceId: string;
+  spaceName: string;
+  pageId: string;
+  dataSourceConfig: DataSourceConfig;
+  isBatchSync: boolean;
+  connectionId: string;
+  connectorId: ModelId;
+}
+
+export async function confluenceUpsertPageActivity({
+  spaceId,
+  spaceName,
+  connectorId,
+  pageId,
+  dataSourceConfig,
+  connectionId,
+  isBatchSync,
+}: ConfluenceUpsertPageActivityInput) {
+  const loggerArgs = {
+    connectorId,
+    dataSourceName: dataSourceConfig.dataSourceName,
+    pageId,
+    spaceId,
+    workspaceId: dataSourceConfig.workspaceId,
+  };
+  const localLogger = logger.child(loggerArgs);
+
+  const confluenceConfig = await fetchConfluenceConfigurationActivity(
+    connectorId
+  );
+
+  const client = await getConfluenceClient({
+    cloudId: confluenceConfig?.cloudId,
+    connectionId,
+  });
+
```

Comment:
Note: for subsequent PR have support for a skip ~here.

Reply:
Done in https://github.com/dust-tt/dust/pull/3328.